pex/pex_bootstrapper.py

```diff
@@ -173,9 +174,22 @@ def maybe_reexec_pex(compatibility_constraints=None):
   os.execv(target_binary, cmdline)
 
 
-def _bootstrap(entry_point):
+def _bootstrap(entry_point, include_pexrc=False, extra_vars=None):
   from .pex_info import PexInfo
   pex_info = PexInfo.from_pex(entry_point)
+
+  # Pull in the default ENV, which is an instance of pex.variables.Variables, which reads
+  # from the default pexrc locations.
+  if include_pexrc:
+    pex_info.update(PexInfo.from_env())
+
+  # Override env settings using extra passed-in vars
+  if extra_vars:
+    if isinstance(extra_vars, dict):
```

Comment:
How about just accepting `**extra_vars` and pushing json back to the user that needs it. It's terrifying that we have a consumer of this API at all, and as a relunctant / conservative library maintainer I'd much rather keep API surface area minimal.

Reply:
I'm not totally sure what you mean here, would you mind clarifying?

Are you saying this should only allow properly formatted json here, and require the user to convert from a dict if necessary?

Reply:
@jsirois, if you could clarify I'm more than happy to update this PR!

Reply:
I just mean support the minimal api, which is allowing the user to pass environment variables via **extra_vars. If code that call this wants to support a json encoding of the env vars that code can decode the json to a dict to pass in as **extra_vars.

Concreteley, just:
```python
def _bootstrap(entry_point, include_pexrc=False, **extra_vars):
  from .pex_info import PexInfo
  pex_info = PexInfo.from_pex(entry_point)
  env = dict(PEX_IGNORE_RCFILES=str(not include_pexrc), **extra_vars)
  pex_info.update(PexInfo.from_env(env))
```

Calling code in an application can do this if it want to unburdening Pex from supporting json in this API in perpetuity:
```python
_bootstrap(entry_point, **json.loads(extra_vars_json_string))
```

And, of course, where this is really important is in the public API function, so the **extra_vars style should propagate up to `bootstrap_pex_env`.